docs/content/troubleshooting/log_status_collect.md

```diff
@@ -1,41 +1,121 @@
+
+## Status and log collection
+To collect and display status and logs related to the different components of IBM® block storage CSI driver, use the Kubernetes commands that are found in this section.
+
+Before you begin collecting logs, create a directory for the logs on the control-plane or bastion server.
+
+For example:
+
+```
+mkdir logs
+```
+Save logs and status reports directly to the created directory by adding in the following string at the end of collection command: `> logs/<log_filename>`.
+
+**Important:** Be sure that the logs cover any relevant timeframes for the specific issues that you are trying to debug when gathering logs from the storage system.
+
+### General status and log collection for IBM Support
+Be sure to run the following steps and copy the output to an external file, when engaging IBM Support and sending log collections.
+
+1. Check the node status.
+    
+    `kubectl get nodes`
+2. Check the CSI driver component status.
+
+    `kubectl get -n <namespace> pod -o wide | grep ibm-block-csi`
+3. Check if the PersistentVolumeClaims (PVCs) are _Bound_.
+
+    `kubectl get pvc`
+
+    - If the PVCs are not in the _Bound_ state collect the events of all unbound PVCs. (See [Log collection for unbound PVCs](#log-collection-for-unbound-pvcs).)
+
+### Log collection for all CSI nodepods and their containers
+
+To collect logs for all nodepods, use the following commands:
+
+    nodepods=`kubectl get -n <namespace> pod -l app.kubernetes.io/component=csi-node --output=jsonpath={.items..metadata.name}`
+    
+    for pod in $nodepods; kubectl logs -n <namespace> $pod -c $container > /tmp/logs_collection/${pod}_${container}.log;done;done
 
-## Status collection for CSI pods, daemonset, and statefulset
 
-`kubectl get all -n <namespace>  -l csi`
+### Log collection for all CSI controller containers
 
-### Log collection for IBM block storage CSI driver controller
+To collect logs for all pods and containers, use the following commands:
+    
+    for container in kubectl get -n default pod ibm-block-csi-controller-0 -o jsonpath='{.spec.containers[*].name}';do kubectl logs -n default ibm-block-csi-controller-0 -c $container > /tmp/logs_collection/ibm-block-csi-controller-0_${container}.log;done
 
+
+### Log collection for CSI operator logs
+To collect CSI operator logs, use the following commands:
+
+    operatorpod=`kubectl get pods --all-namespaces |grep ibm-block-csi-operator|awk '{print $2}'`
+    kubectl logs $operatorpod -n <namespace> > logs/operator
+
+
+### Collecting details of all CSI objects and components
+
+`kubectl describe all -l product=ibm-block-csi-driver -n <namespace>`
+
+For example:
+
+    kubectl describe all -l csi -n <namespace> > logs/describe_ibm-block-csi-driver
+
+
+### Status collection for CSI pods, daemonset, and statefulset
+`kubectl get all -n <namespace>  -l product=ibm-block-csi-driver`
+
+For example:
+
+    kubectl get all -n <namespace> -l csi > get_all_ibm-block-csi-driver
```

Comment:
this does not look to me like an example for this command.
also why should we maintain both the command and its example?

Reply:
it doesn't look like the command because the -l csi should have been -l product=ibm-block-csi-driver. it's updated in the example now as well.
1. it's not uncommon to maintain both a command and an example of implementation.
2. this way we have the command along with an example of where they may be storing the logs. we give them the best practice / option to store it in a logs-type file.

documentation-wise this is not incorrect.
If you want, we can make a note within the section saying that all commands are using the `logs` folder as an example and _only_ keep a single command together with the `>logs/`  for each of them.

Reply:
> only keep a single command together with the >logs/ for each of them
sounds good to me